File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;

        for(int i=0;i<nums.size();i++){

            int fin=target-nums[i];

            if(mp.find(fin)!=mp.end())
            return {mp[fin],i};

            mp[nums[i]]=i;
        }

        vector<int> res;
 
return {};

    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int x = nums[i];
            int rem = target - x;

            if (!map.containsKey(rem))
                map.put(nums[i], i);
            else
                return new int[] { i, map.get(rem) };
        }
        return new int[] {};
    }
}